mediapipe/tasks/c/components/containers/matrix_converter.cc

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2023 The MediaPipe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "mediapipe/tasks/c/components/containers/matrix_converter.h"
+
+#include <cstdlib>
+
+#include "Eigen/Core"
+#include "mediapipe/tasks/c/components/containers/matrix.h"
+
+namespace mediapipe::tasks::c::components::containers {
+
+void CppConvertToMatrix(const Eigen::MatrixXf& in, ::Matrix* out) {
+  out->rows = in.rows();
+  out->cols = in.cols();
+  out->data = new float[out->rows * out->cols];
+
+  // Copy data from Eigen matrix to C-style matrix.
+  // This operation copies the elements sequentially as they appear in the Eigen
+  // matrix's internal storage, regardless of whether it's stored in row-major
+  // or column-major order and ensures the integrity of data during the
+  // transfer.
+  for (int i = 0; i < out->rows * out->cols; ++i) {
```

Comment:
Two comments:
- You should be able to use memcpy now.
- You also need to add a comment in the Matrix type to indicate how data is stored. Please verify, but I believe that Eigen uses column-major (you can verify via unit test).

Reply:
@schmidt-sebastian I've used memcpy now and I think I've already mentioned the Matrix stores the data in column-major order already.